tests/eval_files/trio910.py

```diff
@@ -518,3 +529,12 @@ async def foo_test_return():
 @pytest.fixture()
 async def foo_test_return2():
     return
+
+
+async def foo_comprehension_1():
+    [... for x in range(10) if await foo()]
+
+
+# should error
+async def foo_comprehension_2():
+    [await foo() for x in range(10) if foo()]
```

Comment:
comprehensions are not handled at all, so any `await` inside them will be treated as always being executed. They also weren't handled previously

Reply:
Opened #144